engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -575,18 +504,13 @@ public void notifyComponentRemovalAndEntityDestruction(long entityId, EntityRef
     @Override
     //Todo: be able to save components for entities in any pool
```

Comment:
Yep, this is fixed. I need to go back and review my comments and TODOs and stuff to make sure everything's up to date.